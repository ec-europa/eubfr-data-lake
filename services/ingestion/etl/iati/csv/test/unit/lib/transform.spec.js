/**
 * @jest-environment node
 */

import mapper from '../../../src/lib/transform';
import testRecord from '../../stubs/record.json';

describe(`DG IATI CSV transformer`, () => {
  let result = {};

  beforeAll(() => {
    result = mapper(testRecord);
  });

  test(`Throws an error on empty input`, () => {
    expect(mapper).toThrow();
  });

  test(`Produces correct JSON output structure`, () => {
    expect(result).toMatchSnapshot();
  });

  test(`Project location information is correctly mapped`, () => {
    const firstLocation = result.project_locations[0];
    expect(firstLocation).toMatchSnapshot();
  });
});

describe(`DG IATI transformer handling incomplete or inconsistent data`, () => {
  const partialRecord = {
    Name: 'Bulgaria’s capital expands metro network',
    'Timeframe start': '1388530800',
    'Timeframe end': '1420066800',
    'Project location latitude': '42.73806663',
    'Project location longitude': '23.40014509',
    'Project country(ies)': '',
    'EC’s priorities': '',
    Coordinators: '',
    Partners: '',
    'Related links': '',
  };

  test(`Certain fields should always be present after transformation`, () => {
    const partialResult = mapper(partialRecord);
    expect(partialResult).toMatchSnapshot();
  });
});
