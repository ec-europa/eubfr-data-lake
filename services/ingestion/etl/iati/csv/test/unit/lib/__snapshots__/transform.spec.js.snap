// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DG IATI CSV transformer Produces correct JSON output structure 1`] = `
Object {
  "action": "",
  "budget": Object {
    "eu_contrib": Object {
      "currency": "EUR",
      "raw": "EUR 715440.89",
      "value": 715440.89,
    },
    "funding_area": Array [],
    "mmf_heading": "",
    "other_contrib": Object {
      "currency": "",
      "raw": "",
      "value": 0,
    },
    "private_fund": Object {
      "currency": "",
      "raw": "",
      "value": 0,
    },
    "public_fund": Object {
      "currency": "",
      "raw": "",
      "value": 0,
    },
    "total_cost": Object {
      "currency": "",
      "raw": "",
      "value": 0,
    },
  },
  "call_year": "",
  "description": "ce contrat permet à l'établissement d'utilité publique qui assure les opérations décentraliser indirectes publiques dans la Province orientale de fonctionner",
  "ec_priorities": Array [],
  "media": Array [],
  "programme_name": "",
  "project_id": "c45d1416baf1563c9a23e61b01ce2343",
  "project_locations": Array [
    Object {
      "address": "",
      "centroid": null,
      "country_code": "CD",
      "location": null,
      "nuts": Array [],
      "postal_code": "",
      "region": "",
      "town": "",
    },
  ],
  "project_website": "",
  "related_links": Array [],
  "reporting_organisation": "DEVCO",
  "results": Object {
    "available": "",
    "result": "",
  },
  "status": "",
  "sub_programme_name": "",
  "success_story": "",
  "themes": Array [],
  "third_parties": Array [],
  "timeframe": Object {
    "from": "2011-07-05T00:00:00.000Z",
    "from_precision": "day",
    "to": "2015-02-04T00:00:00.000Z",
    "to_precision": "day",
  },
  "title": "Gestion de l'achat de services et lamélioration du fonctionnement des services de santé dans la Province Orientale",
  "type": Array [],
}
`;

exports[`DG IATI CSV transformer Project location information is correctly mapped 1`] = `
Object {
  "address": "",
  "centroid": null,
  "country_code": "CD",
  "location": null,
  "nuts": Array [],
  "postal_code": "",
  "region": "",
  "town": "",
}
`;

exports[`DG IATI transformer handling incomplete or inconsistent data Certain fields should always be present after transformation 1`] = `
Object {
  "action": "",
  "budget": Object {
    "eu_contrib": Object {
      "currency": "",
      "raw": "",
      "value": 0,
    },
    "funding_area": Array [],
    "mmf_heading": "",
    "other_contrib": Object {
      "currency": "",
      "raw": "",
      "value": 0,
    },
    "private_fund": Object {
      "currency": "",
      "raw": "",
      "value": 0,
    },
    "public_fund": Object {
      "currency": "",
      "raw": "",
      "value": 0,
    },
    "total_cost": Object {
      "currency": "",
      "raw": "",
      "value": 0,
    },
  },
  "call_year": "",
  "description": "",
  "ec_priorities": Array [],
  "media": Array [],
  "programme_name": "",
  "project_id": "d41d8cd98f00b204e9800998ecf8427e",
  "project_locations": Array [],
  "project_website": "",
  "related_links": Array [],
  "reporting_organisation": "",
  "results": Object {
    "available": "",
    "result": "",
  },
  "status": "",
  "sub_programme_name": "",
  "success_story": "",
  "themes": Array [],
  "third_parties": Array [],
  "timeframe": Object {
    "from": null,
    "from_precision": "day",
    "to": null,
    "to_precision": "day",
  },
  "title": "",
  "type": Array [],
}
`;
